src/proto/flwr/proto/recordset.proto

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ==============================================================================
+
+syntax = "proto3";
+
+package flwr.proto;
+
+import "flwr/proto/task.proto";
+
+message Array {
+  string dtype = 1;
+  repeated int32 shape = 2;
+  string stype = 3;
+  bytes data = 4;
+}
+
+message ParametersRecord {
+  repeated string data_keys = 1;
+  repeated Array data_values = 2;
```

Comment:
How about using a ProtoBuf `map<string, Array>` here?

Reply:
`ParametersRecord.data` is of type `OrderedDict`. The `map` in ProtoBuf cannot keep the order, so we have to store keys and values separately.

Reply:
Right! There was no strong reason for `OrderedDict`, but we decided to use it to stay closer to the PyTorch `state_dict` and to make the `Parameters` conversion easier b/c `Parameters` sorts layers by index.

Reply:
Let's keep it as-is